### PR TITLE
HPCC-19068 Refactor ESP session and signature data types

### PR DESCRIPTION
**Pull request by RussWhitehead**
Currently security manager uses a MemoryBuffer to store both the session
token, and the digital signature.  For performance and readability, these
are being refactored to an unsigned for the token and a StringBuffer for
the signature

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@wangkx Please review

File: system/security/shared/SecureUser.hpp [hunk 1]
```diff
@@ -40,8 +40,8 @@ class CSecureUser : implements ISecUser, implements ISecCredentials, public CInt
     StringBuffer    m_Peer;
     SecUserStatus   m_status;
     Owned<IProperties> m_parameters;
-    MemoryBuffer    m_sessionToken;
-    MemoryBuffer    m_signature;
+    unsigned        m_sessionToken;
```
Comment (wangkx) [inline system/security/shared/SecureUser.hpp]: You may set m_sessionToken = 0; here

File: system/security/plugins/htpasswdSecurity/htpasswdSecurity.cpp [hunk 2]
```diff
@@ -158,7 +158,7 @@ class CHtpasswdSecurityManager : public CBaseSecurityManager
 		if (0 == user.length())
 			throw MakeStringException(-1, "htpasswd User name is NULL");
 
-		if (sec_user.credentials().getSessionToken().length())//Already authenticated it token exists
+		if (sec_user.credentials().getSessionToken())//Already authenticated it token exists
```
Comment (wangkx) [inline system/security/plugins/htpasswdSecurity/htpasswdSecurity.cpp]: Also check signature?

File: system/security/LdapSecurity/ldapsecurity.ipp [hunk 3]
```diff
@@ -67,8 +67,8 @@ private:
     StringAttr   m_sudoHost;
     StringAttr   m_sudoCommand;
     StringAttr   m_sudoOption;
-    MemoryBuffer m_sessionToken;//User's ESP session token
-    MemoryBuffer m_signature;//User's digital signature
+    unsigned     m_sessionToken;//User's ESP session token
```
Comment (wangkx) [inline system/security/LdapSecurity/ldapsecurity.ipp]: You may set m_sessionToken = 0; here

**Review by wangkx — COMMENTED** [review]
minor comments

**Comment by RussWhitehead** [comment]
@wangkx Please rereview

**Review by wangkx — APPROVED** [review]
@RussWhitehead looks fine.

**Comment by RussWhitehead** [comment]
@richardkchapman Please merge

**Pull request merged**